package com.task1;
import java.io.*;
import java.util.*;
public class Main {

    public static void main(String[] args) {
        // write your code here

        try {
            int a = 10;
            int b = 0;
            int c = a / b;
        } catch (ArithmeticException e) {
            System.out.println("Causes for Exceptions:");
            System.out.println("1) "+ e.getMessage());
            e.printStackTrace();

        }
        try {
            int[] numbers = new int[3];
            numbers[4] = 45;
            System.out.println(numbers[4]);
        } catch (ArrayIndexOutOfBoundsException e) {
            System.out.println("2) " + e.getMessage());
            e.printStackTrace();

        }
        try {
            String a = "40k";
            System.out.println(Integer.parseInt(a));
        } catch (Exception e) {
            System.out.println("3) "+ e.getMessage());
            e.printStackTrace();
        }
        try {
            Object obj = null;
            obj.hashCode();
        } catch (NullPointerException e) {
            System.out.println("4) "+ e.getMessage());
            e.printStackTrace();

        }

        try {
            FileWriter fileWriter = new FileWriter("file.txt");
            fileWriter.close();
            fileWriter.write("I want eat");
        } catch (IOException e) {
            System.out.println("5) "+ e.getMessage());
            e.printStackTrace();
        }

        finally{
            System.out.println();
            System.out.println("There are Exceptions generated by this program:   ");
            System.out.println();
        }

    }
}
